Make props in documentation easier to parse (#984)

Wattenberger opened the pull request:
I took a stab at making the props in the docs easier to parse - I like when the attributes are very visible, but totally understand if the style is too bold.

![image](https://user-images.githubusercontent.com/3506269/46587756-f91d8c80-ca5e-11e8-85a5-7ddeb981412f.png)

Since props in the docs are just generic headers (h3, h4, h5), I targeted h4 and h5s to prevent changing headers in other parts of the docs. There are a few prop declarations that use `h3`s, which are updated in the second commit :raised_hands:

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/uber/react-vis?pullRequest=993) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/uber/react-vis?pullRequest=993) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/uber/react-vis?pullRequest=993) it.</sub>

Wattenberger commented:
Ah yes, good call! Updated to the underline and moved the blue highlight to code elements following a h4/h5. This also includes inline code elements in the text - I could go either way on the blue highlight in general. I think it's a little loud, but not necessarily too much, plus it gives the docs a bit of personality which I like.

![image](https://user-images.githubusercontent.com/3506269/46591410-950fbe00-ca88-11e8-8354-78fa487d8010.png)

I went through and added new lines around `Type:` and `Default:` statements - I think things are looking great!
